crypto/dsa/eddsa/ed25519.go

```diff
@@ -0,0 +1,51 @@
+package eddsa
+
+import (
+	_ed25519 "crypto/ed25519"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+const (
+	ED25519JWACurve    string = "Ed25519"
+	ED25519AlgorithmID string = ED25519JWACurve
+)
+
+func ED25519GeneratePrivateKey() (jwk.JWK, error) {
+	publicKey, privateKey, err := _ed25519.GenerateKey(rand.Reader)
```

Comment:
It would be better (here and anywhere else) to pass an `io.Reader` for a source of random data, rather than using the global `rand.Reader`. This is generally a good idea in case the user wants to use an alternative random source, but can also make testing easier as they can pass a deterministic random generator in and verify the results are consistent.

Reply:
makes sense @alecthomas. thinking i'll probably go the functional options route e.g.

```go
type ed25519KeyGenOptions struct {
	reader io.Reader
}

type ED25519KeyGenOption func(*ed25519KeyGenOptions)

func WithReader(reader io.Reader) ED25519KeyGenOption {
	return func(args *ed25519KeyGenOptions) {
		args.reader = reader
	}
}

func ED25519GeneratePrivateKey(opts ...ED25519KeyGenOption) (jwk.JWK, error) {
	options := ed25519KeyGenOptions{rand.Reader}
	for _, opt := range opts {
		opt(&options)
	}

	publicKey, privateKey, err := _ed25519.GenerateKey(options.reader)
	if err != nil {
		return jwk.JWK{}, err
	}

	privKeyJwk := jwk.JWK{
		KTY: KeyType,
		CRV: ED25519JWACurve,
		D:   base64.RawURLEncoding.EncodeToString(privateKey),
		X:   base64.RawURLEncoding.EncodeToString(publicKey),
	}

	return privKeyJwk, nil
}
```

will do the same for `generatePrivateKey` in  `secp256k1.go`, `ecdsa.go`, and `eddsa.go`

Reply:
will address this in a separate PR so that we can get this merged